src/v/raft/state_machine_manager.h

```diff
@@ -139,6 +139,31 @@ class state_machine_manager final {
         ss::sstring name;
         stm_ptr stm;
     };
+    /**
+     * Initial recovery snapshot is used by the state machine manager to store
+     * the information about the State Machines initial recovery. This way a
+     * state machine manager can skip to the end the log for the state machines
+     * that require that.
+     */
+    struct initial_recovery_snapshot
+      : serde::envelope<
```

Comment:
Is this data automatically checksummed by the snapshot manager or do we need to use `checksum_envelope`?

Reply:
yes, snapshot is checksummed